tools/serverpod_cli/test/generator/dart/server_code_generator/endpoints_test.dart

```diff
@@ -240,4 +240,50 @@ void main() {
           contains('streamParams[\'streamParam\']!.cast<String>()'));
     });
   });
+
+  group(
+      'Given a protocol definition with a method with "@Deprecated(..)" annotation when generating endpoints file',
+      () {
+    var endpointName = 'testing';
+    var methodName = 'deprecatedMethod';
+    var protocolDefinition = ProtocolDefinition(
+      endpoints: [
+        EndpointDefinitionBuilder()
+            .withClassName('${endpointName.pascalCase}Endpoint')
+            .withName(endpointName)
+            .withMethods([
+          MethodDefinitionBuilder().withName(methodName).withAnnotations([
+            const AnnotationDefinition(
+              name: 'Deprecated',
+              arguments: ["'This method is deprecated.'"],
+              methodCallAnalyzerIgnoreRule:
+                  'deprecated_member_use_from_same_package',
+            )
```

Comment:
Suggestion(Non-Blocking): Add a builder pattern for AnnotationDefinition.

We use a builder pattern for all other definitions. The purpose of that is so that we don't have to change all tests that instantiate the class if we change the signature of the class. This way we always have a solid and valid AnnotationDefinition.